services/job_runner.go

```diff
@@ -68,10 +77,16 @@ func ExecuteRunAtBlock(
 	overrides models.RunResult,
 	bn *models.IndexableBlockNumber,
 ) (models.JobRun, error) {
-	jr.Status = models.RunStatusInProgress
+	if jr.Status == models.RunStatusUnstarted ||
```

Comment:
We shouldn't ever enter this function without the status being unstarted or pending confirmations, right? What do we gain by guarding here? I'm afraid we won't mark it to InProgress, but proceed to update it. Maybe we should exit early if it's not PendingConfirmations/InProgress?

Reply:
We enter this function with an errored job if the amount is insufficient for the chainlink node.